source/slang/ir-glsl-legalize.cpp

```diff
@@ -285,80 +302,121 @@ GLSLSystemValueInfo* getGLSLSystemValueInfo(
         // TODO: deal with `gl_SampleMaskIn` when used as an input.
 
         // TODO: type conversion is required here.
+
+        // uint in hlsl, int in glsl
+        // https://www.opengl.org/sdk/docs/manglsl/docbook4/xhtml/gl_SampleMask.xml
+
+        requiredType = builder->getBasicType(BaseType::Int);
+
         name = "gl_SampleMask";
     }
     else if(semanticName == "sv_depth")
     {
+        // Float in hlsl & glsl
+        // https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/gl_FragDepth.xhtml
         name = "gl_FragDepth";
     }
     else if(semanticName == "sv_depthgreaterequal")
     {
         // TODO: layout(depth_greater) out float gl_FragDepth;
+
+        // Type is 'unknown' in hlsl 
         name = "gl_FragDepth";
     }
     else if(semanticName == "sv_depthlessequal")
     {
         // TODO: layout(depth_greater) out float gl_FragDepth;
+
+        // 'unknown' in hlsl, float in glsl 
         name = "gl_FragDepth";
     }
     else if(semanticName == "sv_dispatchthreadid")
     {
+        // uint3 in hlsl, uvec3 in glsl
+        // https://www.opengl.org/sdk/docs/manglsl/docbook4/xhtml/gl_GlobalInvocationID.xml
         name = "gl_GlobalInvocationID";
 
-        auto builder = context->getBuilder();
         requiredType = builder->getVectorType(builder->getBasicType(BaseType::UInt), builder->getIntValue(builder->getIntType(), 3));
     }
     else if(semanticName == "sv_domainlocation")
     {
+        // TODO: Not 100% confident that say float2 will convert into float3 glsl?
```

Comment:
No, I would expect `adjustType` to have to make the change in vector width explicitly.